### PR TITLE
Change PackingPlan models to use Sets instead of Maps

### PR DESCRIPTION
**Pull request by billonahill**
Using Maps in this API adds unnecessary friction without benefit. This clean-up is the second half of #1323.

See `heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java` for the key changes. The rest of the classes are just updated to reflect the `PackingPlan` API change.

### CODE REVIEW

File: heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java [hunk 1]
```diff
@@ -16,16 +16,24 @@
...
+  private final Map<String, ContainerPlan> containersMap;
+  private final Set<ContainerPlan> containers;
   private final Resource resource;
 
-  public PackingPlan(String id, Map<String, ContainerPlan> containers, Resource resource) {
+  public PackingPlan(String id, Set<ContainerPlan> containers, Resource resource) {
     this.id = id;
     this.containers = containers;
     this.resource = resource;
+    containersMap = new HashMap<>();
```
Comment (ashvina) [inline heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java]: As mentioned in #1324, this map could be a immutable map?

Comment (billonahill) [inline heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java]: That map is initialized in the constructor and never exposed outside the class so making it immutable isn't really necessary. If there was a getter to return it, I would agree.

Comment (ashvina) [inline heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java]: Agree. Perhaps I was looking at this method `getContainers()` which returns the set of containers. The `Set` returned by this method should be immutable?

Comment (billonahill) [inline heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java]: Yeah, I was thinking about that. Updated to copy into an ImmutableSet during initialization.

**Comment by ashvina** [comment]
LGTM 👍

**Comment by maosongfu** [comment]
LGTM :+1:

**Pull request merged**